### PR TITLE
Add filesystem documentation

### LINKED ISSUE #4: Filesystem
Covers files and folders created by EasyEngine. Path and purpose of each.

### PR DESCRIPTION
**Pull request by kirtangajjar**
Closes #4

### CODE REVIEW

File: handbook/filesystem.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## EasyEngine on the Filesystem
+
+All files created by EasyEngine are stored at `/opt/easyengine`
+
+## Sites
+
+By default, all EasyEngine sites are created at `/opt/easyengine/sites`. Configuration and logs of individual site's NGINX, PHP, Postfix are stored in `/opt/easyengine/sites/example.com`. 
+
+Checkout the documentation on [a site's filesystem structure](https://github.com/EasyEngine/docs/blob/master/handbook/site-structure.md) to know more about it.
```
Comment (rahul286) [inline handbook/filesystem.md]: use relative URL for internal linking
```suggestion
Checkout the documentation on [a site's filesystem structure](site-structure.md) to know more about it.
```

File: handbook/filesystem.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+## EasyEngine on the Filesystem
+
+All files created by EasyEngine are stored at `/opt/easyengine`
+
+## Sites
+
+By default, all EasyEngine sites are created at `/opt/easyengine/sites`. Configuration and logs of individual site's NGINX, PHP, Postfix are stored in `/opt/easyengine/sites/example.com`.
```
Comment (rahul286) [inline handbook/filesystem.md]: > default

What is a way to override default?

Comment (rahul286) [reply handbook/filesystem.md]: > Configuration and logs of

are not source code, database and everything relates to a site is stored under `/opt/easyengine/sites/example.com` 🤔

File: handbook/filesystem.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Sites
+
+By default, all EasyEngine sites are created at `/opt/easyengine/sites`. Configuration and logs of individual site's NGINX, PHP, Postfix are stored in `/opt/easyengine/sites/example.com`. 
+
+Checkout the documentation on [a site's filesystem structure](https://github.com/EasyEngine/docs/blob/master/handbook/site-structure.md) to know more about it.
+
+## Services
+
+There are some containers in EasyEngine which serves multiple sites. These are called services. We have following services in EasyEngine:
```
Comment (rahul286) [inline handbook/filesystem.md]: > servers

replace by `are used by`

File: handbook/filesystem.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+1. MariaDB
+2. Nginx Proxy
+3. Cron Scheduler
+
+Files of these services can be found at `/opt/easyengine/services`
```
Comment (rahul286) [inline handbook/filesystem.md]: > Files

Only files? What kind of files?

File: handbook/filesystem.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+SSL Certificates of all sites are stored at - `/opt/easyengine/services/nginx-proxy/certs`.
+
+Each site which has SSL will have 3 files in here - 
+`example.com.crt` - SSL Certificate(Public Key)
+`example.com.key` - Private key
+`example.com.chain.pem` - Certificate Chain File 
+
+#### htpasswd
+
+EasyEngine supports adding basic auth via [auth-command](https://github.com/easyengine/auth-command). The file having details of auth are kept at `/opt/easyengine/services/nginx-proxy/htpasswd`.
```
Comment (mbtamuli) [inline handbook/filesystem.md]: We can use _EasyEngine supports adding HTTP Basic Authentication via_

File: handbook/filesystem.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+Data of MariaDB can be found at `/opt/easyengine/services/mariadb/data`.
+
+### Nginx Proxy
+#### Logs
+Logs of Nginx Proxy can be found at `/opt/easyengine/services/nginx-proxy/logs`.
+
+#### Config
+Config of Nginx Proxy can be found at `/opt/easyengine/services/nginx-proxy/conf.d`.
+
+One part of Nginx Proxy is autogenerated from [go template](https://github.com/EasyEngine/dockerfiles/blob/master/nginx-proxy/nginx.tmpl) and can be found at `/opt/easyengine/services/nginx-proxy/conf.d/default.conf`. Do not modify this file as gets regenerated whenever a container starts/stops.
```
Comment (mbtamuli) [inline handbook/filesystem.md]: This part can be improved. Mainly mentioning why the default.conf is being regenerated, even if we don't give detailed description, at least one/two line should be there.

Comment (mbtamuli) [reply handbook/filesystem.md]: _Do not modify this file as **it** gets regenerated whenever a container starts/stops._

File: handbook/filesystem.md (hunk 2 above)
Comment (mbtamuli) [reply handbook/filesystem.md]: Yes it is.

Comment (kirtangajjar) [reply handbook/filesystem.md]: > What is a way to override default?

Earlier there was a provision to override it. I was under the impression it was still there. Looks like it was removed.

**Pull request merged**